File: leetcode/medium/medium.539/medium.539.go
package medium539

import (
	"strconv"
	"strings"
	"sort"
)

func findMinDifference(timePoints []string) int {
	totalMinutesInDay := 1440
	minDiff := totalMinutesInDay
	minutePoints := []int{}
	
	for _, time := range timePoints {
		parts := strings.Split(time, ":")
		hours, _ := strconv.Atoi(parts[0])
		minutes, _ := strconv.Atoi(parts[1])
		totalMinutes := hours * 60 + minutes

		minutePoints = append(minutePoints, totalMinutes)
	}

	sort.Ints(minutePoints)

	for i := 1; i < len(minutePoints); i++ {
		diff := minutePoints[i] - minutePoints[i-1]
		if diff < minDiff {
			minDiff = diff
		}
	}

	circularDiff := totalMinutesInDay - minutePoints[len(minutePoints) - 1] + minutePoints[0]
	if circularDiff < minDiff {
		minDiff = circularDiff
	}

	return minDiff
}

File: leetcode/medium/medium.539/medium.539.py
from typing import List

class Solution:
    @classmethod
    def findMinDifference(self, timePoints: List[str]) -> int:
        totalMinutesInDay = 1440
        minDiff = totalMinutesInDay
        minutePoints = []
        
        for time in timePoints:
          hours, minutes = time.split(':')
          totalMinutes = int(hours) * 60 + int(minutes)
          minutePoints.append(totalMinutes)
          
        minutePoints = sorted(minutePoints)

        for i in range (1, len(minutePoints)):
          diff = minutePoints[i] - minutePoints[i - 1]
          minDiff = min(minDiff, diff)
          
        circularDiff = totalMinutesInDay - minutePoints[-1] + minutePoints[0]
        minDiff = min(minDiff, circularDiff)
          
        return minDiff
      
result1 = Solution.findMinDifference(["23:59", "00:00"])
print(result1)